### DISCUSSION TITLE
forkMode not handling default branch change

### DISCUSSION THREAD

**Discussion by enocom** [renovatebot/renovate#8844] [Request Help] [answered]
Thanks for all the hard work on renovate. As an open source maintainer, I find it incredibly helpful, but have a minor configuration problem at the moment.

Just yesterday following [the docs][0] I added `:semanticCommitTypeAll(chore)` to the `extends` section of  our [`renovate.json`][1], hoping to see all future PRs prefixed with `chore`, but this morning I received [a PR with a `fix` prefix][2].

Have I misconfigured something?

Here's the entire config:

``` json
{
  "extends": [
    "config:base",
    ":semanticCommitTypeAll(chore)"
  ],
  "prConcurrentLimit": 0,
  "rebaseStalePrs": true,
  "masterIssue": true,
  "semanticCommits": true,
  "postUpdateOptions": [
    "gomodTidy"
  ]
}
```
[0]: https://github.com/renovatebot/renovate/blob/master/docs/usage/semantic-commits.md#changing-the-semantic-commit-type
[1]: https://github.com/GoogleCloudPlatform/cloudsql-proxy/blob/59bf84ff80f634df78ecbe1f1000019ec576c080/.github/renovate.json#L4
[2]: https://github.com/GoogleCloudPlatform/cloudsql-proxy/pull/632

**Comment by HonkingGoose** [comment]
EDIT: Turns out using `:semanticPrefixChore` won't fix the issue.

~I think `:semanticPrefixChore` would get you what you want?~

Quote from docs:

> :semanticPrefixChore
> Use `chore` as semantic commit type for commit messages and PR titles

See the docs here: https://docs.renovatebot.com/presets-default/#semanticprefixchore

---

Also I wanted to mention that `masterIssue` is a old term, we now use `dependencyDashboard` to do the same thing.

You can still use `masterIssue` as Renovate migrates old-terms behind the scenes anyways.
But if you search for `masterIssue` on the documentation pages for Renovate, you will not get any hits.
To prevent confusion for future-you, you might want to replace `masterIssue` with `dependencyDashboard`.

**Reply by enocom** [reply]
Fantastic! Thank you! And I'm glad to see we can use `dependencyDashboard` instead. I'll make the change.

**Reply by rarkins** [reply]
Sorry but that won't fix it. Please see my answer in the thread below:

**Reply by HonkingGoose** [reply]
> Sorry but that won't fix it. Please see my answer in the thread below:

Oh well, it was worth a try.

Ignore this silly Goose then. 😄 🙃

**Comment by rarkins** [comment]
I would have thought that `":semanticCommitTypeAll(chore)"` *should* work. `config:base` is opinionated on semantic prefixes (that's where the "fix" comes from) so whatever you configure needs to override that fully, or disable the sub-preset with `ignorePresets`

**Reply by HonkingGoose** [reply]
@rarkins You can see the PR that's intended to fix the `chore` commit/pr prefix problem here: https://github.com/GoogleCloudPlatform/cloudsql-proxy/pull/633
Might want to take a look and confirm that the new config will fix their issue?

**Reply by rarkins** [reply]
In the meantime the workaround would be to add `:semanticPrefixFixDepsChoreOthers` to `ignorePresets`:

```json
{
  "extends": [
    "config:base",
    ":semanticCommitTypeAll(chore)"
  ],
  "ignorePresets": [":semanticPrefixFixDepsChoreOthers"],
  "prConcurrentLimit": 0,
  "rebaseStalePrs": true,
  "masterIssue": true,
  "semanticCommits": true,
  "postUpdateOptions": [
    "gomodTidy"
  ]
}
```

**Reply by HonkingGoose** [reply]
@rarkins Want me to open a `docs:` issue to discuss how to deal with the `ignorePresets` thing in the Semantic Commits docs?

**Reply by enocom** [reply]
Thanks @rarkins. I really appreciate it.

**Comment by shubha-rajan** [comment]
Following up on this. The semantic commit messages are still showing up as `fix`. We're also seeing these errors on our dependency dashboard:
![image](https://user-images.githubusercontent.com/19631367/109337519-ca640480-7819-11eb-8216-202338e3c495.png)

I'm curious if it has to do with the fact that our `renovate.json` isn't in the root directory

**Reply by rarkins** [reply]
Please reproduce in a public repo. There's a good chance it's a config issue

**Reply by enocom** [reply]
This is coming from https://github.com/GoogleCloudPlatform/cloudsql-proxy (a public project).

**Reply by enocom** [reply]
Renovate config is here: https://github.com/GoogleCloudPlatform/cloudsql-proxy/blob/main/.github/renovate.json.

**Comment by rarkins** [comment, marked as answer]
Looking at the repo's logs, the reason appears to be that the default branch switched from `master` to `main` while the forked repo has not. So the config is being read from the old `master` branch in the fork still.

We added code to try to manage this transition automatically but our API calls are getting rejected. I've attempted to fix it in this commit which I'll deploy later today: https://github.com/renovatebot/renovate/commit/f4e9e92e77bd60f37186f2319f94b4dc16bda084

**Reply by rarkins** [reply]
Now resolved

**Reply by enocom** [reply]
Thank you!
